src/utils/healthcheck.ts

```diff
@@ -46,9 +51,9 @@ const toHealthProblems = <S extends ProblemSource>(source: S) => (
  *
  * @returns either true or an array of error messages
  */
-export const checkConfigHealth = <IConfig>(ConfigCodec: t.Type<IConfig>) => (
-  value: unknown
-): HealthCheck<"Config", IConfig> =>
+export const checkConfigHealth = <IConfig extends t.Mixed>(
+  ConfigCodec: IConfig
+) => (value: unknown): HealthCheck<"Config", t.OutputOf<IConfig>> =>
```

Comment:
Can you explain a bit about the rationale behind your refactoring? Just to make it clear.

Reply:
Actually, thanks to your comment, I thought about it again, and I decided to change the return type from 
```
(value: unknown): HealthCheck<"Config", t.OutputOf<IConfig>>
```

to 
```
(value: unknown): HealthCheck<"Config", t.TypeOf<IConfig>>
```

The Idea is to return a function that takes an unknown value and returns Either a "Config" error or the decodec value.
We don't need the Output encoded value, but the internal one.